test/object-store/sync/session/progress_notifications.cpp

```diff
@@ -591,27 +589,91 @@ TEST_CASE("progress notification", "[sync]") {
             callback_was_called_2 = false;
             current_uploaded = 36;
             current_uploadable = 310;
-            current_downloaded = 250;
-            current_downloadable = 50;
+            current_downloaded = 182;
+            current_downloadable = 190;
             progress.update(current_downloaded, current_downloadable, current_uploaded, current_uploadable, 1, 1);
             CHECK(callback_was_called);
             CHECK(transferred == current_downloaded);
-            CHECK(transferrable == 250);
+            CHECK(transferrable == original_downloadable);
             CHECK(callback_was_called_2);
             CHECK(downloaded == current_downloaded);
-            CHECK(downloadable == 250);
+            CHECK(downloadable == original_downloadable_2);
 
             // Third callback, last one for second notifier
             callback_was_called = false;
             callback_was_called_2 = false;
             current_uploaded = 36;
             current_uploadable = 310;
-            current_downloaded = 250;
-            current_downloadable = 0;
+            current_downloaded = 189;
+            current_downloadable = 250;
             progress.update(current_downloaded, current_downloadable, current_uploaded, current_uploadable, 1, 1);
             CHECK(!callback_was_called);
-            CHECK(!callback_was_called_2);
+            CHECK(callback_was_called_2);
             CHECK(downloaded == current_downloaded);
+            CHECK(downloadable == original_downloadable_2);
+
+            // Fourth callback
+            callback_was_called_2 = false;
+            current_uploaded = 36;
+            current_uploadable = 310;
+            current_downloaded = 201;
+            current_downloadable = 289;
+            progress.update(current_downloaded, current_downloadable, current_uploaded, current_uploadable, 1, 1);
+            CHECK(!callback_was_called_2);
+        }
+
+        SECTION("download notifiers handle transferrable decreasing") {
```

Comment:
I'm sure that there's a test case that would have caught this issue and that it somehow slipped through CR, but could you point out to me where it is in this file?

Reply:
The tests *did* catch that there was a functional change, and then the tests were modified to pass.